activerecord/lib/active_record/relation/query_methods.rb

```diff
@@ -1160,8 +1162,7 @@ def check_if_method_has_arguments!(method_name, args)
 
     def structurally_compatible_for_or?(other)
       Relation::SINGLE_VALUE_METHODS.all? { |m| send("#{m}_value") == other.send("#{m}_value") } &&
-        (Relation::MULTI_VALUE_METHODS - [:extending]).all? { |m| send("#{m}_values") == other.send("#{m}_values") } &&
-        (Relation::CLAUSE_METHODS - [:having, :where]).all? { |m| send("#{m}_clause") != other.send("#{m}_clause") }
+        (Relation::MULTI_VALUE_METHODS - [:extending]).all? { |m| send("#{m}_values") == other.send("#{m}_values") }
```

Comment:
Was the `Relation::CLAUSE_METHODS - [:having, :where])` check obviated or somehow unnecessary?